### ISSUE TITLE
Recomputation trigerred although function didn't change

### ISSUE THREAD

**Issue by mblondel** [joblib/joblib#324]
Sometimes, re-computation of a cached function is triggered if I change parts of the file the function is in, although I didn't change the function itself. So, I was wondering what algorithm do you use to decide whether to re-compute a function or not? Do you also take into account the function context?

**Comment by lesteve** [comment]
I seem to remember people reporting the same behaviour, @AlexandreAbraham for example. I took it as a slightly annoying but conservative joblib behaviour.

I am unable to reproduce this on a toy example though, do you have one ?

**Comment by mblondel** [comment]
Nope, I am unable to recreate a simple example but this has annoyed me a few times.
